tessel/src/lib.rs

```diff
@@ -82,86 +78,146 @@ pub struct PortGroup {
 /// use tessel::Port;
 /// ```
 pub struct Port {
-    // Path of the domain socket.
     socket: Arc<Mutex<PortSocket>>,
-    pins: HashMap<usize, Mutex<()>>,
 }
 
-pub struct Pin<'a> {
-    index: usize,
-    _guard: MutexGuard<'a, ()>,
-    socket: Arc<Mutex<PortSocket>>,
-}
+impl Port {
+    pub fn new(path: &str) -> Port {
+        // Create and return the port struct
+        Port {
+            socket: Arc::new(Mutex::new(PortSocket::new(path))),
+        }
+    }
 
-impl<'a> Pin<'a> {
-    pub fn output(&mut self, value: bool) -> io::Result<()> {
-        let mut sock = self.socket.lock().unwrap();
-        if value {
-            sock.write_command(Command::GpioHigh(self.index as u8))
-        } else {
-            sock.write_command(Command::GpioLow(self.index as u8))
+    pub fn pins(&mut self) -> (Pin, Pin, Pin) {
+        (
+            Pin::new(5, self.socket.clone()),
+            Pin::new(6, self.socket.clone()),
+            Pin::new(7, self.socket.clone()),
+        )
+    }
+
+    pub fn i2c<'b>(self) -> (I2cPort<'b>, Gpio<'b>) {
+        let mut available = BTreeSet::new();
```

Comment:
Also, small nitpick that `available` doesn't seem like the best name. Perhaps `pins_available` would be more clear?